File: src/factoryMethod/Playground.java
package factoryMethod;

import factoryMethod.factory.Blacksmith;
import factoryMethod.factory.elve.ElvenBlacksmith;
import factoryMethod.factory.orc.OrcishBlacksmith;
import factoryMethod.weapon.WeaponTypes;

public class Playground {
    public static void main(String[] args) {

        Blacksmith elveBlacksmith  = new ElvenBlacksmith();
        Blacksmith orcishBlacksmith  = new OrcishBlacksmith();

        WeaponTypes spear = WeaponTypes.Spear;
        System.out.println(elveBlacksmith.create(spear).getName());
        System.out.println(orcishBlacksmith.create(spear).getName());

        System.out.println();

        WeaponTypes sword = WeaponTypes.Sword;
        System.out.println(elveBlacksmith.create(sword).getName());
        System.out.println(orcishBlacksmith.create(sword).getName());

    }
}

File: src/factoryMethod/abstractFactory/Playground.kt
package factoryMethod.abstractFactory

import factoryMethod.abstractFactory.blacksmith.Blacksmith
import factoryMethod.abstractFactory.factory.ElvenWeaponFactory
import factoryMethod.abstractFactory.factory.OrcishWeaponFactory
import factoryMethod.abstractFactory.weapon.WeaponType

fun main() {
    val elvenWeaponFactory = ElvenWeaponFactory()
    val orcishWeaponFactory = OrcishWeaponFactory()

    val orchBlacksmith = Blacksmith(orcishWeaponFactory)
    val elvenBlacksmith = Blacksmith(elvenWeaponFactory)

    println(orchBlacksmith.createWeapon(WeaponType.Sword).name)
    println(orchBlacksmith.createWeapon(WeaponType.Axe).name)
    println()
    println(elvenBlacksmith.createWeapon(WeaponType.Sword).name)
    println(elvenBlacksmith.createWeapon(WeaponType.Axe).name)
}